watcher.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/go-logr/logr"
+)
+
+type telegrafClassesOnChange func()
+
+// telegrafClassesWatcher allows monitoring a directory with telegraf classes using
+// fsnotify package and batching multiple events to reduce number of Kubernetes API calls.
+type telegrafClassesWatcher struct {
+	logger        logr.Logger
+	watcherEvents chan fsnotify.Event
+	onChange      telegrafClassesOnChange
+
+	eventCount   uint64
+	eventChannel chan bool
+	eventDelay   time.Duration
+}
+
+// newTelegrafClassesWatcher creates a new instance of telegrafClassesWatcher.
+func newTelegrafClassesWatcher(logger logr.Logger, telegrafClassesDirectory string, onChange telegrafClassesOnChange) (*telegrafClassesWatcher, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	// watching the contents of classes directory requires adding the directory as well as most child elements
+	logger.Info("adding directory to watcher", "directory", telegrafClassesDirectory)
+	watcher.Add(telegrafClassesDirectory)
+
+	items, err := ioutil.ReadDir(telegrafClassesDirectory)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, item := range items {
+		name := item.Name()
+
+		// add all items in the classes directory except for current/previous secret contents that begin with "..", "." and ".."
```

Comment:
I see why "." and ".." are to be ignored; but why should we ignore all other files that begin with ".."?
I have a vague memory that the k8s secrets mount contains a lot of special files and symlinks etc; it would help if you render make an example in the comment or point to such an example in some public doc.

Reply:
Very good point. Added more comments.

I learned an interesting lesson here - since this was in a livestream, I explained it in details to the audience, but did not capture it in a comment.